phy/cluster/manual/wizard.py

```diff
@@ -67,62 +99,395 @@ def set_quality_function(self, func):
+
+    def _delete(self, clusters):
+        for clu in clusters:
+            if clu in self._cluster_groups:
+                del self._cluster_groups[clu]
+            if clu in self._best_list:
+                self._best_list.remove(clu)
+            if clu in self._match_list:
+                self._match_list.remove(clu)
+
+    def _add(self, clusters, group):
+        for clu in clusters:
+            self._cluster_groups[clu] = group
+            if self.best is not None:
+                index = self._best_list.index(self.best)
+                self._best_list.insert(index, clu)
+
+    def move(self, cluster, group):
+        self._cluster_groups[cluster] = group
+        if cluster == self.best:
+            self.next_best()
+            # self._set_match_list(cluster)
+        elif cluster == self.match:
+            self.next_match()
+        # if self.match is not None:
+        #     self._set_match_list()
+        self._check()
+
+    def merge(self, old, new, group):
+        if isinstance(new, (tuple, list)):
+            assert len(new) == 1
+            new = new[0]
+        # Add new cluster.
+        self._add([new], group)
+        # Delete old clusters.
+        self._delete(old)
+        # Pin the newly-created cluster.
+        if self.best in old and self.match is not None and self.match in old:
+            self.pin(new)
+        # TODO: the match list is not updated currently; it could be if
+        # necessary.
+        self._check()
+
+    def update_clusters(self, deleted, added, group):
+        self._delete(deleted)
+        self._add(added, group)
+        # Update the best cluster if it was deleted.
+        if self.best in deleted and self.match is not None:
+            self.pin(added[0])
+        self._check()
+
+    # Panel
+    #--------------------------------------------------------------------------
+
+    @property
+    def _best_progress(self):
+        """Progress in the best clusters."""
+        value = self.best_list.index(self.best)
+        maximum = len(self.best_list)
+        return _progress(value, maximum)
+
+    @property
+    def _match_progress(self):
+        """Progress in the processed clusters."""
+        value = self.n_processed
+        maximum = self.n_clusters
+        return _progress(value, maximum)
+
+    def _repr_html_(self):
+        return _wizard_panel_html(best=self.best,
+                                  match=self.match,
+                                  best_progress=self._best_progress,
+                                  match_progress=self._match_progress,
+                                  best_group=self._group(self.best),
+                                  match_group=self._group(self.match),
+                                  )
+
+
+#------------------------------------------------------------------------------
+# Wizard panel
+#------------------------------------------------------------------------------
+
+_PANEL_HTML = """
```

Comment:
this should probably be in a separate .html and .css file, like the vert/frag shaders.